lambda/index.js

```diff
@@ -8,13 +8,25 @@ const Sharp = require('sharp');
 
 const BUCKET = process.env.BUCKET;
 const URL = process.env.URL;
+const ALLOWED_RESOLUTIONS = process.env.ALLOWED_RESOLUTIONS ? new Set(process.env.ALLOWED_RESOLUTIONS.split(/\s*,\s*/)) : new Set([]);
 
 exports.handler = function(event, context, callback) {
   const key = event.queryStringParameters.key;
-  const match = key.match(/(\d+)x(\d+)\/(.*)/);
-  const width = parseInt(match[1], 10);
-  const height = parseInt(match[2], 10);
-  const originalKey = match[3];
+  const match = key.match(/((\d+)x(\d+))\/(.*)/);
+
+  //Check if requested resolution is allowed
+  if(0 != ALLOWED_RESOLUTIONS.size && !ALLOWED_RESOLUTIONS.has(match[1]) ) {
```

Comment:
Could this be phrased in the affirmative for easier parsing? I'd suggest something like:

```javascript
if(ALLOWED_DIMENSIONS.size > 0 && ALLOWED_DIMENSIONS.has(dimensions)) {
  // Do the thing.
}
```

I'd also like to see the matches set to locals for easier parsing up top, so:

```javascript
const dimensions = match[1];
```

Reply:
No problem with this one.